[ubuntu-precompiled] add support for the KERNEL_MODULE_TYPE env var

tariq1890 opened the pull request:
As of gpu-operator v25.3.0, we no longer set the `OPEN_KERNEL_MODULES_ENABLED` variable and it is replaced by the `KERNEL_MODULE_TYPE` variable. This PR brings the precompiled containers to parity with the runfile ones

cdesiniotis commented on ubuntu22.04/precompiled/nvidia-driver:
```diff
@@ -232,6 +237,45 @@ _unload_driver() {
 
+_fetch_nvidia_installer () {
+    curl -fSsl -O $BASE_URL/$DRIVER_VERSION/$DRIVER_RUN_FILE.run
+    chmod +x  $DRIVER_RUN_FILE.run
```
minor nit: 
```suggestion
    chmod +x $DRIVER_RUN_FILE.run
```

tariq1890 replied:
Thanks

cdesiniotis commented on ubuntu22.04/precompiled/nvidia-driver:
```diff
@@ -276,6 +320,10 @@ _unmount_rootfs() {
 }
 
 init() {
+    _fetch_nvidia_installer
```
Why are we doing this at runtime? Shouldn't we be fetching the `nvidia-installer` at build time and including the binary in the image? Ideally the precompiled image should be entirely self-contained.

tariq1890 replied:
Agreed. I'll move this to the build time

cdesiniotis commented on ubuntu24.04/precompiled/nvidia-driver:
```diff
@@ -232,6 +237,45 @@ _unload_driver() {
 
+_fetch_nvidia_installer () {
+    curl -fSsl -O $BASE_URL/$DRIVER_VERSION/$DRIVER_RUN_FILE.run
+    chmod +x  $DRIVER_RUN_FILE.run
```
minor nit: 
```suggestion
    chmod +x $DRIVER_RUN_FILE.run
```

tariq1890 commented:
@cdesiniotis I've tested the changes after addressing your review comments. Also checked the contents of the pulled image `ghcr.io/nvidia/driver:abf7755f-570-6.8.0-56-generic-ubuntu24.04` and confirmed that `nvidia-installer` was present 

Requesting another round of review
